consensus/src/chained_bft/block_storage/block_tree.rs

```diff
@@ -211,6 +211,18 @@ where
     pub(super) fn insert_quorum_cert(&mut self, qc: QuorumCert) -> Result<(), BlockTreeError> {
         let block_id = qc.certified_block_id();
         let qc = Arc::new(qc);
+
+        // Safety invariant: For any two quorum certificates qc1, qc2 in the block store,
+        // qc1 == qc2 || qc1.round != qc2.round
+        // The check below establishes the above invariant for every incoming
+        // quorum certificate(qc), thus making it a linear time check.
+        assume!({
```

Comment:
btw, every `checked_` might introduce a crash if we're not careful! I can imagine a wrong `assume` followed by a `checked_` for example.

Reply:
Changing the `assume` to `debug_checked_precondition` caused a bunch of test cases to fail. These tests were creating `placeholder QCs` for newly created blocks even when the `parent QC` was known causing the invariant to break. I have fixed them in my latest commit. Thanks!

Reply:
btw @DavidLDill just told me about something similar so I thought it was interesting. We probably, most of the time, shouldn't rewrite a specification/annotation because the code is complaining, we should re-write the code to comply with the specification instead :)